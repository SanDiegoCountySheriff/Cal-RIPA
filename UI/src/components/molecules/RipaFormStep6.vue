<template>
  <v-form ref="stepForm" lazy-validation>
    <ripa-agency-questions
      v-model="model"
      :on-open-statute="onOpenStatute"
    ></ripa-agency-questions>

    <v-spacer></v-spacer>

    <template v-if="!isFormValid">
      <ripa-alert alert-type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </ripa-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <template v-if="backButtonVisible">
        <v-btn
          outlined
          color="primary"
          class="tw-mr-2"
          :disabled="isBackNextDisabled"
          @click="handleBack"
        >
          Back
        </v-btn>
      </template>
      <v-btn outlined color="error" class="tw-mr-2" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn color="primary" :disabled="isBackNextDisabled" @click="handleNext">
        Next
      </v-btn>
    </div>
  </v-form>
</template>

<script>
import RipaAlert from '@/components/atoms/RipaAlert'
import RipaAgencyQuestions from '@/components/molecules/RipaAgencyQuestions'
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step6',

  mixins: [RipaFormStepMixin],

  components: {
    RipaAlert,
    RipaAgencyQuestions,
  },

  props: {
    backButtonVisible: {
      type: Boolean,
      default: true,
    },
  },
}
</script>
