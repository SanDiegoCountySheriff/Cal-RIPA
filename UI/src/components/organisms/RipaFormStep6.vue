<template>
  <v-form ref="stepForm" lazy-validation>
    <div>REVIEW FORM SELECTIONS</div>
    <v-spacer></v-spacer>

    <template v-if="!isValid">
      <v-alert type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </v-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <v-btn outlined color="primary" class="tw-mr-2" @click="handleBack">
        Back
      </v-btn>
      <v-btn outlined color="error" class="tw-mr-2" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn color="primary" @click="handleSubmit"> Submit </v-btn>
    </div>
  </v-form>
</template>

<script>
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step6',

  mixins: [RipaFormStepMixin],
}
</script>
