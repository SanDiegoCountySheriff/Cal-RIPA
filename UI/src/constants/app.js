export const VERSION = '#{TEMPLATE_VERSION}#'
