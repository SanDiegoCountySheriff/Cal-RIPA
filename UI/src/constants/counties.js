export const COUNTIES = [
  {
    name: 'Alameda',
  },
  {
    name: 'Alpine',
  },
  {
    name: 'Amador',
  },
  {
    name: 'Butte',
  },
  {
    name: 'Calaveras',
  },
  {
    name: 'Colusa',
  },
  {
    name: 'Contra Costa',
  },
  {
    name: 'Del Norte',
  },
  {
    name: 'El Dorado',
  },
  {
    name: 'Fresno',
  },
  {
    name: 'Glenn',
  },
  {
    name: 'Humboldt',
  },
  {
    name: 'Imperial',
  },
  {
    name: 'Inyo',
  },
  {
    name: 'Kern',
  },
  {
    name: 'Kings',
  },
  {
    name: 'Lake',
  },
  {
    name: 'Lassen',
  },
  {
    name: 'Los Angeles',
  },
  {
    name: 'Madera',
  },
  {
    name: 'Marin',
  },
  {
    name: 'Mariposa',
  },
  {
    name: 'Mendocino',
  },
  {
    name: 'Merced',
  },
  {
    name: 'Modoc',
  },
  {
    name: 'Mono',
  },
  {
    name: 'Monterey',
  },
  {
    name: 'Napa',
  },
  {
    name: 'Nevada',
  },
  {
    name: 'Orange',
  },
  {
    name: 'Placer',
  },
  {
    name: 'Plumas',
  },
  {
    name: 'Riverside',
  },
  {
    name: 'Sacramento',
  },
  {
    name: 'San Benito',
  },
  {
    name: 'San Bernardino',
  },
  {
    name: 'San Diego',
  },
  {
    name: 'San Francisco',
  },
  {
    name: 'San Joaquin',
  },
  {
    name: 'San Luis Obispo',
  },
  {
    name: 'San Mateo',
  },
  {
    name: 'Santa Barbara',
  },
  {
    name: 'Santa Clara',
  },
  {
    name: 'Santa Cruz',
  },
  {
    name: 'Shasta',
  },
  {
    name: 'Sierra',
  },
  {
    name: 'Siskiyou',
  },
  {
    name: 'Solano',
  },
  {
    name: 'Sonoma',
  },
  {
    name: 'Stanislaus',
  },
  {
    name: 'Sutter',
  },
  {
    name: 'Tehama',
  },
  {
    name: 'Trinity',
  },
  {
    name: 'Tulare',
  },
  {
    name: 'Tuolumne',
  },
  {
    name: 'Ventura',
  },
  {
    name: 'Yolo',
  },
  {
    name: 'Yuba',
  },
]
