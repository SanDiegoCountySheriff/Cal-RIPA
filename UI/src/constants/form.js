export const OFFICER_ASSIGNMENTS = [
  { name: 'Patrol, traffic enforcement, field operations', value: 1 },
  { name: 'Gang enforcement', value: 2 },
  { name: 'Compliance check', value: 3 },
  { name: 'Special events', value: 4 },
  { name: 'Roadblock or DUI sobriety checkpoint', value: 5 },
  { name: 'Narcotics/vice', value: 6 },
  { name: 'Task force', value: 7 },
  { name: 'K-12 public school', value: 8 },
  { name: 'Investigative/detective', value: 9 },
  { name: 'Other', value: 10 },
]

export const DURATIONS = [
  { name: '0 - 10', value: 1 },
  { name: '11 - 20', value: 2 },
  { name: '21 - 30', value: 3 },
  { name: '31 - 60', value: 4 },
  { name: '60+', value: 5 },
]

export const AGES = [
  { name: '0 - 10', value: 1 },
  { name: '11 - 20', value: 2 },
  { name: '21 - 30', value: 3 },
  { name: '31 - 60', value: 4 },
  { name: '60+', value: 5 },
]

export const DISABILITIES = [
  { name: 'Deafness or difficulty hearing', value: 1 },
  { name: 'Speech impairment or limited use of language', value: 2 },
  { name: 'Blind or limited vision', value: 3 },
  { name: 'Mental health condition', value: 4 },
  {
    name: 'Intellectual or developmental disability, including dementia',
    value: 5,
  },
  { name: 'Other disability', value: 6 },
]

export const GENDERS = [
  { name: 'Male', value: 1 },
  { name: 'Female', value: 2 },
  { name: 'Transgender Male', value: 3 },
  { name: 'Transgender Female', value: 4 },
  { name: 'Gender Noncomforming', value: 5 },
]

export const RACES = [
  { name: 'Asian', value: 1 },
  { name: 'Black/African American', value: 2 },
  { name: 'Hispanic/Latino(a)', value: 3 },
  { name: 'Middle Eastern or South Asian', value: 4 },
  { name: 'Native American', value: 5 },
  { name: 'Pacific Islander', value: 6 },
  { name: 'White', value: 7 },
]

export const STOP_REASONS = [
  { name: 'Traffic Violation', value: 1 },
  { name: 'Reasonable Suspicion', value: 2 },
  {
    name: 'Known to be on Parole / Probation / PRCS / Mandatory Supervision',
    value: 3,
  },
  {
    name: 'Knowledge of outstanding arrest warrant/wanted person',
    value: 4,
  },
  {
    name: 'Investigation to determine whether the person was truant',
    value: 5,
  },
  { name: 'Consensual Encounter resulting in a search', value: 6 },
]

export const TRAFFIC_VIOLATIONS = [
  { name: 'Moving Violation', value: 1 },
  { name: 'Equipment Violation', value: 2 },
  {
    name: 'Non-moving Violation, including Registration Violation',
    value: 3,
  },
]

export const REASONABLE_SUSPICIONS = [
  { name: 'Officer witnessed commission of a crime', value: 1 },
  { name: 'Matched suspect description', value: 2 },
  {
    name: 'Witness or Victim identification of Suspect at the scene',
    value: 3,
  },
  { name: 'Carrying Suspicious Object', value: 4 },
  {
    name: 'Actions indicative of casing a victim or location',
    value: 5,
  },
  { name: 'Suspected of Acting as Lookout', value: 6 },
  { name: 'Actions indicative of drug transaction', value: 7 },
  {
    name: 'Actions indicative of engaging in violent crime',
    value: 8,
  },
  { name: 'Other Reasonable Suspicion of a crime', value: 9 },
]

export const ACTIONS_TAKEN_GENERAL = [
  { name: 'Person removed from vehicle by order', value: 1 },
  { name: 'Person removed from vehicle by physical contact', value: 2 },
  { name: 'Field sobriety test conducted', value: 3 },
  { name: 'Curbside detention', value: 4 },
  { name: 'Handcuffed or flex cuffed', value: 5 },
  { name: 'Patrol car detention', value: 6 },
  { name: 'Canine removed from vehicle or used to search', value: 7 },
  { name: 'Firearm pointed at person', value: 8 },
  { name: 'Firearm discharged or used', value: 9 },
  { name: 'Electronic control device used', value: 10 },
  { name: 'Impact projectile discharged or used', value: 11 },
  { name: 'Canine bit or held person', value: 12 },
  { name: 'Baton or other impact weapon used', value: 13 },
  { name: 'Chemical spray used', value: 14 },
  { name: 'Physical or vehicle contact', value: 15 },
  { name: 'Personal photographed', value: 16 },
  { name: 'Vehicle impounded', value: 22 },
]

export const ACTIONS_TAKEN_SEARCH = [
  { name: 'Asked for consent to search another person', value: 17 },
  { name: 'Search of person was conducted', value: 18 },
  { name: 'Asked for consent to search property', value: 19 },
  { name: 'Search of property was conducted', value: 20 },
]

export const BASIS_FOR_PROPERTY_SEIZURE = [
  { name: 'Safekeeping as allowed by law/statute', value: 1 },
  { name: 'Contraband', value: 2 },
  { name: 'Evidence', value: 3 },
  { name: 'Impound of vehicle', value: 4 },
  { name: 'Impound of vehicle', value: 5 },
]

export const PROPERTY_SEIZED_TYPES = [
  { name: 'Firearm(s)', value: 1 },
  { name: 'Ammunition', value: 2 },
  { name: 'Weapon(s) other than a firearm', value: 3 },
  { name: 'Drugs/narcotics', value: 4 },
  { name: 'Alcohol', value: 5 },
  { name: 'Money', value: 6 },
  { name: 'Drug paraphernalia', value: 7 },
  { name: 'Suspected stolen property', value: 8 },
  { name: 'Cell phone(s) or electronic device(s)', value: 9 },
  { name: 'Vehicle', value: 10 },
  { name: 'Other contraband or evidence', value: 11 },
]
