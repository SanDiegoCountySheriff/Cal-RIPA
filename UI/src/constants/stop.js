export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
