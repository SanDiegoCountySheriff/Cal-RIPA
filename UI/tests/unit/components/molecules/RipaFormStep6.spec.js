import RipaFormStep6 from '@/components/molecules/RipaFormStep6.vue'
import { STOP } from '../../constants/RipaFormContainerTestConstants'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Form Step 6', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaFormStep6, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaFormStep6, {
      vuetify,
      propsData: {
        value: STOP,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
