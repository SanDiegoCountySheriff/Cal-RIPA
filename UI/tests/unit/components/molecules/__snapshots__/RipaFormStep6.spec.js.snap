// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Form Step 6 should match snapshot 1`] = `
<v-form-stub
  lazyvalidation="true"
>
  <ripa-agency-questions-stub
    onopenstatute="[Function]"
    value="[object Object]"
  />
   
  <v-spacer-stub />
   
  <!---->
   
  <div
    class="tw-flex tw-mt-8 tw-justify-center"
  >
    <v-btn-stub
      activeclass=""
      class="tw-mr-2"
      color="primary"
      outlined="true"
      tag="button"
      type="button"
    >
      
        Back
      
    </v-btn-stub>
     
    <v-btn-stub
      activeclass=""
      class="tw-mr-2"
      color="error"
      outlined="true"
      tag="button"
      type="button"
    >
      
      Cancel
    
    </v-btn-stub>
     
    <v-btn-stub
      activeclass=""
      color="primary"
      tag="button"
      type="button"
    >
      
      Next
    
    </v-btn-stub>
  </div>
</v-form-stub>
`;
